crates/ruma-api-macros/src/request/outgoing.rs

```diff
@@ -177,7 +177,7 @@ impl Request {
                 #ruma_serde::json_to_buf(&RequestBody { #initializers })?
             }
         } else {
-            quote! { <T as ::std::default::Default>::default() }
+            quote! { #ruma_serde::slice_to_buf(b"{}") }
         };
```

Comment:
This should be as simple as

```suggestion
        } else if method == "GET" {
            quote! { <T as ::std::default::Default>::default() }
        } else {
            quote! { #ruma_serde::slice_to_buf(b"{}") }
        }
```